src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -110,17 +115,35 @@ export class ItemEditComponent implements OnDestroy, PendingChangesComponent {
     );
   }
 
+  private getItemChanges(): ItemChanges {
+    const urlFormControl = this.itemForm.get('url');
+    const usesApiFormControl = this.itemForm.get('uses_api');
+    const textIdFormControl = this.itemForm.get('text_id');
+
+    if (urlFormControl === null || usesApiFormControl === null || textIdFormControl === null) return {};
+
+    const itemFormValues = {
+      url: urlFormControl.value as string | null,
+      uses_api: usesApiFormControl.value as boolean,
+      text_id: textIdFormControl.value as string,
+    };
+
+    if (itemFormValues.url === '') itemFormValues.url = null;
+
+    return itemFormValues;
+  }
+
   private updateItem(): Observable<void> {
     return this.createChildren().pipe(
       switchMap(res => {
-        const changes: ItemChanges = {};
+        if (!this.initialFormData) return throwError(new Error('Invalid form'));
+        const changes: ItemChanges = this.getItemChanges();
         if (res) {
+          // @TODO: Avoid affecting component vars in Observable Operator
           // save the new children (their ids) to prevent recreating them in case of error
           this.itemChanges.children = res;
           changes.children = res.map((child, idx) => ({ item_id: child.id, order: idx }));
         }
-        if (!this.initialFormData) return throwError(new Error('Invalid form'));
-        if (!Object.keys(changes).length) return of(undefined);
```

Comment:
I don't get why you moved L122... but that is not very important.
I don't get why you removed L123, which is problematic as it was preventing updates of nothing. Basically, we don't want to call the update-item service if just the title (changed via update-item-strings) has been changed.

Reply:
because at one point my ItemChanges definition had uses_api as required so it was never empty. But it's not the case anymore. It will be fix by an other PR review fix